geode-core/src/main/java/org/apache/geode/cache/query/security/UnrestrictedMethodAuthorizer.java

```diff
@@ -0,0 +1,120 @@
+ */
+public final class UnrestrictedMethodAuthorizer implements MethodInvocationAuthorizer {
+  static final String NULL_CACHE_MESSAGE = "Cache should be provided to configure the authorizer.";
+  static final String NULL_AUTHORIZER_MESSAGE =
+      "RestrictedMethodAuthorizer should be provided to create this authorizer.";
+  private static final String GEODE_BASE_PACKAGE = "org.apache.geode";
+  private final RestrictedMethodAuthorizer restrictedMethodAuthorizer;
+
+  /**
+   * Creates a {@code UnrestrictedMethodAuthorizer} object and initializes it so it can be safely
+   * used in a multi-threaded environment.
+   * <p/>
+   *
+   * Applications can use this constructor as part of the initialization for custom authorizers
+   * (see {@link Declarable#initialize(Cache, Properties)}), when using a declarative approach.
+   *
+   * @param cache the {@code Cache} instance that owns this authorizer, required in order to
+   *        configure the default {@link RestrictedMethodAuthorizer}.
+   */
+  public UnrestrictedMethodAuthorizer(Cache cache) {
+    Objects.requireNonNull(cache, NULL_CACHE_MESSAGE);
+    this.restrictedMethodAuthorizer = new RestrictedMethodAuthorizer(cache);
+  }
+
+  /**
+   * Creates a {@code UnrestrictedMethodAuthorizer} object and initializes it so it can be safely
+   * used in a multi-threaded environment.
+   * <p/>
+   *
+   * @param restrictedMethodAuthorizer the default {@code RestrictedMethodAuthorizer} to use.
+   */
+  public UnrestrictedMethodAuthorizer(RestrictedMethodAuthorizer restrictedMethodAuthorizer) {
+    Objects.requireNonNull(restrictedMethodAuthorizer, NULL_AUTHORIZER_MESSAGE);
+    this.restrictedMethodAuthorizer = restrictedMethodAuthorizer;
+  }
+
+  /**
+   * Executes the authorization logic to determine whether the {@code method} is allowed to be
+   * executed on the {@code target} object instance.
+   * If the {@code target} object is an instance of {@link Region}, this methods also ensures that
+   * the user has the {@code DATA:READ} permission granted for the target {@link Region}.
+   * <p/>
+   *
+   * @param method the {@link Method} that should be authorized.
+   * @param target the {@link Object} on which the {@link Method} will be executed.
+   * @return {@code true} if the {@code method} can be executed on on the {@code target} instance,
+   *         {@code false} otherwise.
+   *
+   * @see org.apache.geode.cache.query.security.MethodInvocationAuthorizer
+   */
+  @Override
+  public boolean authorize(Method method, Object target) {
+    // Return false for known dangerous methods.
+    if (restrictedMethodAuthorizer.isKnownDangerousMethod(method, target)) {
+      return false;
+    }
+
+    // Return true for non Geode classes.
+    String packageName = target.getClass().getPackage().getName().toLowerCase();
+    if (!packageName.startsWith(GEODE_BASE_PACKAGE)) {
+      return true;
+    }
+
+    // Delegate to the default authorizer.
```

Comment:
This comment is true but it's just delegating to the restricted authorizer just for the geode methods portion.  not sure if that was the intended intent of the comment or not.

Reply:
Yes, this was intended: the execution can only reach this line whenever the actual `target` objects belongs to Geode, so we need the determine whether that Geode method is allowed or not (through `restrictedMethodAuthorizer.isAllowedGeodeMethod`):

```
    // Return true for non Geode classes.
    String packageName = target.getClass().getPackage().getName().toLowerCase();
    if (!packageName.startsWith(GEODE_BASE_PACKAGE)) {
      return true;
    }

    // Delegate to the default authorizer.
    return restrictedMethodAuthorizer.isAllowedGeodeMethod(method, target);
```